### PR TITLE
docs

### PR DESCRIPTION
**Pull request by justinmk**
todo

- [x] fix `:trust` doc. [ref](https://github.com/neovim/neovim/commit/6752f1005d26c93a033d856a60b7b296f3e51634#r101699399)

### CODE REVIEW

File: src/nvim/main.c [hunk 1]
```diff
@@ -2171,43 +2171,33 @@ static void usage(void)
...
+  os_msg(_("  nvim [options] [file ...]\n"));
   os_msg(_("\nOptions:\n"));
-  os_msg(_("  --                    Only file names after this\n"));
-  os_msg(_("  +                     Start at end of file\n"));
   os_msg(_("  --cmd <cmd>           Execute <cmd> before any config\n"));
   os_msg(_("  +<cmd>, -c <cmd>      Execute <cmd> after config and first file\n"));
   os_msg(_("  -l <script> [args...] Execute Lua <script> (with optional args)\n"));
+  os_msg(_("  -S <session>          Source <session> after loading the first file\n"));
+  os_msg(_("  -s <scriptin>         Read Normal mode commands from <scriptin>\n"));
+  os_msg(_("  -u <config>           Use this config file\n"));
```
Comment (justinmk) [inline src/nvim/main.c:2181]: Group all "execute" modes at the top.

File: src/nvim/main.c [hunk 2]
```diff
@@ -2171,43 +2171,33 @@ static void usage(void)
...
   os_msg("\n");
-  os_msg(_("  -b                    Binary mode\n"));
   os_msg(_("  -d                    Diff mode\n"));
-  os_msg(_("  -e, -E                Ex mode\n"));
   os_msg(_("  -es, -Es              Silent (batch) mode\n"));
   os_msg(_("  -h, --help            Print this help message\n"));
   os_msg(_("  -i <shada>            Use this shada file\n"));
-  os_msg(_("  -m                    Modifications (writing files) not allowed\n"));
-  os_msg(_("  -M                    Modifications in text not allowed\n"));
```
Comment (justinmk) [inline src/nvim/main.c:2191]: Remove uncommon flags from `nvim -h`. They can be found in `man nvim` and `:help startup-options`.

**Comment by dundargoc** [comment]
I *hate* that github doesn't allow pointing to code that wasn't touched by the PR.

Anyway, there's a line in `dev_style.txt` that we concluded was kind of silly and should be removed: "Do not use filenames that already exist in `/usr/include`, such as `db.h`." I forgot to remove it since it was so small, but you could do it in this PR perhaps.

**Comment by justinmk** [comment]
> "Do not use filenames that already exist in `/usr/include`, such as `db.h`."

Why is it silly? That idea was recently applied in e.g. https://github.com/neovim/neovim/pull/22442/commits/b68c020a9c7f51b905bf57c4051ab5198efc66b0

**Comment by dundargoc** [comment]
Both you and Björn was against this rule the last time we discussed it, with the rationale "that is too open ended, the user or their distro could put any filename there." I was going to remove it from the style guide then, I just forgot to. Anyway, I didn't mean to introduce an unrelated, impromptu discussion about this, I just assumed the concensus was still the same. Sorry for the noise.

**Comment by bfredl** [comment]
It is mostly relevant for system headers we actually use (because uncrustify otherways gets into a fit, like in the referenced commit), but it is good style anyway to avoid libc/POSIX header names I guess.

**Comment by dundargoc** [comment]
nvim_create_user_command is missing the nargs parameter

**Comment by dundargoc** [comment]
`MAINTAIN.md` might needs some adjustments if you want to take a look.

>E.g. the `+plan` label increases the ticket's priority merely
for having a plan written down: it is _closer to completion_ than tickets
without a plan.

Did we use to have a `+plan` label previously that got removed? Or is this
referring to the `has:plan` label? Just got a bit confused.

> `src/nvim/lib/`

klib is now under `src/klib`

> treesitter parsers

Do these count as vendored? The queries I'd understand, but these just seem
like normal dependencies?

> We may maintain forks, if we are waiting on upstream changes

Ugh, could we maybe drop this? Maintaining a fork is the nuclear option IMO,
the sentence is a bit too casual. cmake already allows us to patch things if
needed in a pinch.

---

My own addition: should we add a note about experimental features on release
since there's been a few of them (cmdheight=0 and statuscolumn comes to mind)?
If this is something the neovim team wants to embrace I think it could be worth
a mention.

Disclaimer: I'm not *personally* a fan of marking features as experimental, but
I think it could be worth explicitly mentioning the neovim team's stance on
this regardless.

File: runtime/doc/insert.txt [hunk 3]
```diff
@@ -1081,6 +1081,26 @@ Stop completion						*compl-stop*
...
+  api.nvim_create_autocmd("InsertCharPre", {
+    buffer = api.nvim_get_current_buf(),
+    callback = function()
+      if tonumber(vim.fn.pumvisible()) == 1 then
+        return
+      end
+      local char = vim.v.char
+      if vim.tbl_contains(triggers, char) then
+        local key = vim.keycode("<C-x><C-n>")
+        api.nvim_feedkeys(key, "m", true)
```
Comment (zeertzjq) [inline runtime/doc/insert.txt:1098]:
```suggestion
        api.nvim_feedkeys(key, "m", false)
```

File: runtime/doc/insert.txt [hunk 4]
```diff
@@ -1081,6 +1081,26 @@ Stop completion						*compl-stop*
...
 
+AUTO-COMPLETION						*compl-autocomplete*
+
+To get basic "autocompletion" without installing a plugin, try this script: >lua
+
+  local triggers = {"."}
+  api.nvim_create_autocmd("InsertCharPre", {
+    buffer = api.nvim_get_current_buf(),
+    callback = function()
+      if tonumber(vim.fn.pumvisible()) == 1 then
```
Comment (zeertzjq) [inline runtime/doc/insert.txt:1092]:
```suggestion
      if vim.fn.pumvisible() == 1 then
```

**Pull request merged**